os/fs/driver/mtd/mtd_partition.c

```diff
@@ -795,7 +785,7 @@ FAR struct mtd_dev_s *mtd_partition(FAR struct mtd_dev_s *mtd, off_t firstblock,
 	part->tagnumber = tagno;
 
 #ifdef CONFIG_MTD_PARTITION_NAMES
-	part->name = NULL;
+	strcpy(part->name, "(noname)");
```

Comment:
You'd better using strncpy instead of strcpy.

Reply:
I modified it.